pkg/server/api/agent/v1/service_test.go

```diff
@@ -51,3 +156,110 @@ func setupServiceTest(t *testing.T) *serviceTest { //nolint: unused,deadcode
 
 	return test
 }
+
+func (s *serviceTest) createTestNodes(ctx context.Context, t *testing.T) {
+	for _, testNode := range testNodes {
+		// create the test node
+		_, err := s.ds.CreateAttestedNode(ctx, &datastore.CreateAttestedNodeRequest{Node: testNode})
+		require.NoError(t, err)
+
+		// set selectors to the test node
+		_, err = s.ds.SetNodeSelectors(ctx, &datastore.SetNodeSelectorsRequest{Selectors: testNodeSelectors[testNode.SpiffeId]})
+		require.NoError(t, err)
+	}
+}
+
+func TestGetAgent(t *testing.T) {
+	for _, tt := range []struct {
+		name          string
+		req           *agentpb.GetAgentRequest
+		expectedAgent *types.Agent
+		expectedCode  codes.Code
+		err           string
+		logMsg        string
+		dsError       error
+	}{
+		{
+			name:          "success 1",
```

Comment:
what is the diference between success 1 2 and 3?

Reply:
can you add a test case where no selectors found?

Reply:
I'll remove agent2.